### ISSUE TITLE
Websockets and reloading: one subscription - two open requests upon reload

### ISSUE THREAD

**Issue by rundis** [fluxxu/elm-hot-loader#15]
I have a simple app with a websocket subscription. When I reload the app, the connection is closed (which is reasonable I would suppose), but then my websocket server receives two open requests rather than one. The next time I reload I get a close event for the first of the two from the previous round, followed by another 2 open requests. Over time this ends up with a bunch of sockets (which I can easily see in the network tab in devtools). This is obviously problematic :-) 
Of course I can't be sure it's due to elm-hot-loader,  but looking at the behavior it seems it's not entirely improbable either ?
#### webpack-config.js

``` JavaScript
devServer: {
    stats: 'errors-only',
     proxy: {
      '/ws/*': {
        target: 'ws://localhost:3333/ws',
        ws: true,
      },
    },
  }
```
#### Main.elm

``` haskell

type Msg
    = ServerMsg String
    | SavePredictions String

subscriptions : Model -> Sub Msg
subscriptions model =
    WebSocket.listen "ws://localhost:3333/ws" ServerMsg

update : Msg -> Model -> ( Model, Cmd Msg )
update msg model = 
        ServerMsg msg ->
            let
                _ = Debug.log "Msg from server" msg
            in
                ( model, Cmd.none )

        SavePredictions str ->
            (model, WebSocket.send "ws://localhost:3333/ws" str)


view : Model -> Html.Html Msg
view model =
    div [ class "container" ]
        [ div [class "row"]
             [ button 
                 [ class "btn btn-success"
                 , onClick (SavePredictions "Scorepredictions coming soonish!")
               ] 
               [text "Send WS message"] 
            ]
```

**Comment by rundis** [comment]
Seems to be sort of reproducible with http://guide.elm-lang.org/architecture/effects/web_sockets.html too

**Comment by fluxxu** [comment]
I think this is because elm-lang/websocket will attempt to reconnect after socket close:
https://github.com/elm-lang/websocket/blob/master/src/WebSocket.elm#L262

Not 100% sure, but I think it's solvable...
Will release a patch soon.

Thanks!

**Comment by fluxxu** [comment]
Just published 0.3.3 😄
